dataclients/kubernetes/definitions/routegroups.go

```diff
@@ -121,7 +122,7 @@ type skipperBackendParser struct {
 
 type BackendReference struct {
 	// BackendName references the skipperBackend by name
-	BackendName string `json:"backendName"`
+	BackendName string `json:"backendName" yaml:"backendName"`
```

Comment:
Interesting why is this the only field that was making a problem

Reply:
K8s decided not to support yaml tags as that is a hassle to have a tag per each serialization format. Instead they decided to convert yaml to json first and then use json deserializer, see https://pkg.go.dev/sigs.k8s.io/yaml#section-readme for more details.

For us this means that definitions.ParseRouteGroupsYAML (again not used anywhere) is broken. OTOH since we know it is broken it means noone uses it so we can remove.

Reply:
To load your testcase without adding yaml tags you can use ghodss/yaml that we import as a dependency.
I am also not sure how existing yaml test fixtures work without yaml tag and `(sb *SkipperBackend) UnmarshalYAML`, please check.

Reply:
> not sure how existing yaml test fixtures work without yaml tag

Because we load json
https://github.com/zalando/skipper/blob/master/dataclients/kubernetes/clusterclient.go#L355-L359

Reply:
I think currently we have 2 directions

- Use json in the test
- Try to create the test under `testFixtures` stuff to also use k8s dataclient
- Fix Yaml marshalling

I would stick with first option

Reply:
> Because we load json

but fixtures are yamls :)